packages/util/src/orderBook/index.ts

```diff
@@ -0,0 +1,209 @@
+      const orderBook = key.args[0].toJSON();
+      const meta = value.unwrap();
+      const { orderBookId, status, lastOrderId, tickSize, stepLotSize, minLotSize, maxLotSize } = meta;
+      const base = orderBookId.toJSON().base;
+      const quote = orderBookId.toJSON().quote;
+
+      buffer[toKey(base) + toKey(quote)] = {
+        orderBook,
+        status: status.toString(),
+        lastOrderId: lastOrderId.toNumber(),
+        tickSize: this.getValueFromJson(tickSize as unknown as Value),
+        stepLotSize: this.getValueFromJson(stepLotSize as unknown as Value),
+        minLotSize: this.getValueFromJson(minLotSize as unknown as Value),
+        maxLotSize: this.getValueFromJson(maxLotSize as unknown as Value),
+      };
+
+      return buffer;
+    }, {});
+
+    this.orderBooks = orderBooks;
+  }
+
+  /**
+   * Get user's order book addresses and set to public userOrderBooks array
+   *
+   */
+  public async getUserOrderBooks(account: string): Promise<void> {
+    const toKey = (address) => address.code.toString();
+
+    const userOrderBooksIds = [];
+
+    const entries = await this.root.api.query.orderBook.userLimitOrders.entries(account);
+    entries.map(([book]) => {
+      userOrderBooksIds.push(toKey(book.toHuman()[1].base) + toKey(book.toHuman()[1].quote));
+    });
+
+    this.userOrderBooks = userOrderBooksIds;
+  }
+
+  /**
+   * Creates a subscription that relies on block emitting
+   * @param account account address
+   * @returns array of book addresses the user is in
+   */
+  public subscribeOnUserOrderBooks(account: string): Observable<Promise<String[]>> {
+    return this.root.system.getBlockNumberObservable().pipe(
+      map(async () => {
+        await this.getUserOrderBooks(account);
+        return this.userOrderBooks;
+      })
+    );
+  }
+
+  /**
+   * Get mappings price to amount of asks
+   * @param orderBookId base and quote addresses
+   */
+  public getAggregatedAsks(base: string, quote: string): Observable<Array<[]>> {
+    return this.root.apiRx.query.orderBook.aggregatedAsks({ dexId: 0, base, quote }).pipe(
+      map((data) => {
+        const priceAmountMapping = data.toJSON() as any;
+        let asks = [];
+
+        for (const ask of Object.keys(priceAmountMapping)) {
+          const price = this.getValue(JSON.parse(ask));
+          const amount = this.getValue(priceAmountMapping[ask]);
+
+          asks.push([price, amount]);
+        }
+
+        return asks;
+      })
+    );
+  }
+
+  /**
+   * Get mappings price to amount of bids
+   * @param orderBookId base and quote addresses
+   */
+  public getAggregatedBids(base: string, quote: string): Observable<Array<[]>> {
+    return this.root.apiRx.query.orderBook.aggregatedBids({ dexId: 0, base, quote }).pipe(
+      map((data) => {
+        const priceAmountMapping = data.toJSON() as any;
+        let bids = [];
+
+        for (const bid of Object.keys(priceAmountMapping)) {
+          const price = this.getValue(JSON.parse(bid));
```

Comment:
Why JSON.parse?

Reply:
because general-purpose function expects this input, it's used in many other places